### PR TITLE
Added Selection sort and Insertion sort algorithm

### PR DESCRIPTION
**Pull request by youknowqyh**

### CODE REVIEW

File: pydatastructs/linear_data_structures/algorithms.py [hunk 1]
```diff
@@ -1507,6 +1650,83 @@ def binary_search(array, value, **kwargs):
...
+        middle = left//2 + right//2 + left % 2 * right % 2
+        if array[middle] == value:
+                return middle
+        if comp(array[middle], value):
+            left = middle + 1
+        else:
+            right = middle - 1
+
+    return None
+
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/algorithms.py:1729]:
```suggestion
```

**Comment by czgdp1807** [comment]
Please make sure you pull the changes from `sortalgo` (your branch for this PR) before pushing back anything.

**Comment by youknowqyh** [comment]
> Please make sure you pull the changes from `sortalgo` (your branch for this PR) before pushing back anything.

Sure!

**Comment by youknowqyh** [comment]
Is there any issue about this?

**Comment by czgdp1807** [comment]
I will merge this on Sunday after pushing some polishing commits to your branch.

**Comment by youknowqyh** [comment]
> I will merge this on Sunday after pushing some polishing commits to your branch.

Thanks for your guiding! I'll continue reading the source code and contribute something else in this project :)

**Comment by czgdp1807** [comment]
> I'll continue reading the source code and contribute something else in this project

It would be great if you can take a look at the open issues as well. There is a lot of stuff we need help with.

**Comment by youknowqyh** [comment]
> > I'll continue reading the source code and contribute something else in this project
> 
> It would be great if you can take a look at the open issues as well. There is a lot of stuff we need help with.

Yeah sure!

**Comment by czgdp1807** [comment]
Thanks @youknowqyh

**Pull request merged**